test/integration/control.js

```diff
@@ -0,0 +1,35 @@
+var path = require('path');
+var test = require('tap').test;
+var extract = require('../fixtures/extract');
+var VirtualMachine = require('../../src/index');
+
+var uri = path.resolve(__dirname, '../fixtures/control.sb2');
+var project = extract(uri);
+
+test('control project', function (t) {
+    var vm = new VirtualMachine();
+
+    // Evaluate playground data and exit
+    vm.on('playgroundData', function (e) {
+        var threads = JSON.parse(e.threads);
+        t.ok(threads.length > 0);
+        t.end();
+        process.nextTick(process.exit);
```

Comment:
`tap` executes each test file in a separate process, but contrary to what you may expect `t.end()` does not exit the process (regardless of how it is called). Rather the Node.js process will stay open as long as something is keeping it running (e.g. a singleton, an interval timer, etc.). In the case of the VM, once you `vm.start()` an interval timer is created which will prevent the Node.js from exiting. One way to resolve this would be to build up some `vm.halt()` infrastructure, the other is to simply use `process.nextTick(process.exit)` which will force the process to exit regardless of the state of the program. This is fairly normal and something we use in other test suites for some of our other repos.